### ISSUE TITLE
[Bug]: doorbell_pressed push is attributed to the wrong space on a multi-space PRO account

### ISSUE THREAD

**Issue by Sven2410** [bvis/aegis-hass#494]
### Describe the bug

First, a thank you: the Video Edge entities I reported earlier are fixed in 1.19.1.
They are back and updating.

New issue. I use a PRO account with several client spaces. Per-space selection works
well, but doorbell events do not respect the space boundary: pressing the doorbell at
one location fires the ring event on a doorbell entity belonging to a different
location.

Three things stand out in the debug log:

1. `group_id=None` on every doorbell push. If no group or space is carried, attribution
   has nothing to distinguish one space's doorbell from another's.
2. `notification_id` is byte-for-byte identical across three presses that were roughly
   7 minutes and 1.5 minutes apart. If that ID is meant to identify a single event, it
   is being reused. As a side effect, the 5s duplicate window would also swallow two
   genuinely different presses close together.
3. `extracted_device_id` and the resolved device are constant, while the press was
   physically at another location.

This may also explain something I saw on 1.17: my own doorbell fired ring events
roughly five times per minute while nobody was pressing it. I assumed a push storm at
the time; it now looks like it could have been presses from other spaces landing on my
entity.

### Steps to reproduce

Configure the integration with a PRO account that has access to more than one space,
   and select more than one space.
2. Enable debug logging for custom_components.aegis_ajax.
3. Press the doorbell in space A.
4. The push is parsed with group_id=None and attributed to a device that is not the one
   pressed. The ring event fires on the doorbell entity of the wrong space.

### Expected behavior

A doorbell push is attributed to the space and the device it originated from. A ring in
space A fires only the event entity of the doorbell in space A, and never triggers a
doorbell entity belonging to another client's space.

### Home Assistant version

2026.9.1

### Integration version

1.19.1

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text
Three doorbell presses at a location other than the one they resolved to.
IDs and device references redacted; identical placeholders mean identical values
in the original.

22:02:55 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-1>
22:02:55 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:02:55 [aegis_ajax.notification] Push event parsed: event_type=doorbell_pressed raw_tag=ring_button_pressed group_id=None
22:02:55 [aegis_ajax.notification] Push device attribution: event_type=doorbell_pressed extracted_device_id=<DEVICE-ID> device_type=VIDEO_EDGE resolved=<RESOLVED-DEVICE>
22:02:55 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-2>
22:02:55 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:02:55 [aegis_ajax.notification] Duplicate notification_id <NOTIF-ID> within 5s window; skipping fire/refresh

22:09:42 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-3>
22:09:42 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:09:42 [aegis_ajax.notification] Push event parsed: event_type=doorbell_pressed raw_tag=ring_button_pressed group_id=None
22:09:42 [aegis_ajax.notification] Push device attribution: event_type=doorbell_pressed extracted_device_id=<DEVICE-ID> device_type=VIDEO_EDGE resolved=<RESOLVED-DEVICE>
22:09:42 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-4>
22:09:42 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:09:42 [aegis_ajax.notification] Duplicate notification_id <NOTIF-ID> within 5s window; skipping fire/refresh

22:11:03 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-5>
22:11:03 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:11:03 [aegis_ajax.notification] Push event parsed: event_type=doorbell_pressed raw_tag=ring_button_pressed group_id=None
22:11:03 [aegis_ajax.notification] Push device attribution: event_type=doorbell_pressed extracted_device_id=<DEVICE-ID> device_type=VIDEO_EDGE resolved=<RESOLVED-DEVICE>
22:11:04 [aegis_ajax.notification] Push notification received: persistent_id=0:<PUSH-ID-6>
22:11:04 [aegis_ajax.notification] Extracted notification_id: <NOTIF-ID>
22:11:04 [aegis_ajax.notification] Duplicate notification_id <NOTIF-ID> within 5s window; skipping fire/refresh
```

### Additional context

The doorbell is an Ajax video doorbell bridged through an Ajax NVR, which is why the
push carries device_type=VIDEO_EDGE.

Unrelated noise in the same log window, in case it comes up: my local DNS was briefly
unreachable around 21:56–21:58, which caused the FCM client to shut down after three
connection errors and restart five minutes later. Push recovered on its own and this
is not part of the report.

Also unrelated but perhaps useful: at startup HA's ONVIF integration logs "No
registered handler for event" for the Ajax RuleEngine ObjectDetector topic from the
same camera family. Different integration, mentioning it only for context.

I run three client installations through a PRO account and can test against real
hardware — Video Edge channels, an NVR and a doorbell — if you want to try a fix.

**Comment by bvis** [comment]
Thank you for this one. The report did most of the diagnostic work for me: all three of the things you flagged as odd are real, and two of them are separate bugs in our code rather than anything Ajax is doing. Both are fixed in #497, merged.

**Your point 1, the wrong entity.** When a doorbell push arrives without a usable source device id, we fell back to "the only doorbell in the install". That list was built from every device the config entry knows about, across every space — so an account with one registered doorbell between several client spaces attributed *every* unattributed ring to that doorbell. A press at one client fires the ring on another client's entity, which is exactly what you saw, and it is why `resolved` was constant in your log while the press was physically elsewhere. The fallback now only considers doorbells paired to the hub of the space the push was actually delivered to, and refuses to guess at all when the push cannot be routed and you have more than one space. A single-space install keeps the old behaviour.

This is the same shape as #358, and for the same reason: per-space behaviour is invisible with one space, so it survives until someone with a PRO account and several clients looks at it. That is the second time your kind of setup has caught one of these.

**Your point 2, the reused notification id.** Ajax is not reusing it. We were reading it wrong. The extractor scanned the payload for the first 64-character hex run instead of reading `Notification.id` structurally. That happens to be the real id on most pushes, but not when the real id is not 64 hex characters — and a video/NVR-sourced push is one such shape, where the scan then latches onto some other blob that is constant per device. Which is precisely "byte-for-byte identical across three presses". It is now read structurally, with the old scan kept only as a fallback for payloads that cannot be decoded at all.

You were also right about the consequence, before I had looked at the code: that value keys the 5-second duplicate window, so a constant made two genuinely different presses look like one delivery and silently dropped the second. Both halves of that are gone with the same fix.

**Your point 3** was the symptom of point 1, and your `group_id=None` observation is expected rather than a fault — a doorbell ring is not a group event, so there is no group to name. Routing has to come from the space, which is what the fix keys on.

**About the 1.17 storm.** Your reading is plausible and I am not going to assert it. Other spaces' presses landing on your entity would look exactly like that, but it is a claim about an older version, and I would rather see it not come back than declare it solved. If it reappears on the build below, that is a separate and interesting datum.

**What I need from you, whenever convenient.** This is the part nobody here can do: reproducing it needs a multi-space PRO account with a doorbell, and I have neither. So it ships as a beta first, and your confirmation is what moves it to stable.

1. Take the next beta when I cut it (HACS → Aegis → three-dot → Redownload → show beta versions). I will comment here with the version number.
2. Press the doorbell at one location and check the ring fires on *that* location's entity and on no other.
3. With debug logging still on, check that `Extracted notification_id` now differs between two presses at the same doorbell a few minutes apart.
4. If you can, two presses within about five seconds of each other — both should now arrive rather than the second being swallowed.

If the ring lands correctly but the id is still constant, that is the more interesting outcome and I would want that log. Either way, please leave debug on for it.

The two unrelated notes you added were both worth adding, and neither is a problem: the DNS blip and FCM restart is the client recovering as designed, and the ONVIF "no registered handler" line belongs to the ONVIF integration and does not touch anything we do. Thank you for separating them out rather than leaving me to guess.

**Comment by bvis** [comment]
`1.20.0-beta.1` is published and available, so this is ready for you whenever you have a moment.

To install it: HACS → Aegis → the three-dot menu → Redownload → enable "show beta versions" → pick `1.20.0-beta.1`, then restart Home Assistant. Keep `custom_components.aegis_ajax: debug` on.

What would settle it, in order of how much it tells me:

1. Press the doorbell at one location. The ring should fire on that location's entity and on no other.
2. Two presses at the same doorbell a few minutes apart: `Extracted notification_id` should now differ between them.
3. If it is easy, two presses within about five seconds. Both should arrive now rather than the second being swallowed.

A "worked, nothing else to report" is a complete answer. If the ring lands correctly but the identifier is still constant, that is the more interesting outcome and the log for it is what I would want.

One thing to expect and not be alarmed by: on an account with several spaces, a ring the integration cannot route to a space is now deliberately left unattributed rather than guessed onto a device. It still fires the hub-level event entity, so an automation on that keeps working. If you see a doorbell that used to fire and now does not, that is this change and I want to know, because it would mean the routing needs the same treatment as the attribution.

**Comment by bvis** [comment]
Reopening: I closed this by accident. The fix PR carried a "Fixes #494" line, which GitHub acted on when it merged.

This should stay open until you have confirmed it on `1.20.0-beta.1`, which is the whole point of shipping it as a beta — nobody here can reproduce the misattribution, so your test is the only evidence there will be. I will close it when you report back, or if it turns out the fix is incomplete, it stays open and gets another round.

**Comment by Sven2410** [comment]
@bvis worked, nothing else to report!

**Comment by bvis** [comment]
Perfect! 👏 . I'm closing this, thanks for your contribution!
